### ISSUE TITLE
ApiGatewayResolver root routes and strip_prefixes

### ISSUE THREAD

**Issue by BVMiko** [awslabs/aws-lambda-powertools-python#646]
I ran into an issue when using a route with the rule `"/"` along with `strip_prefixes` feature.  This rule is unable to trigger due to the `path` variable being reduced down to an empty string instead of a forward slash. 

```python
from aws_lambda_powertools.event_handler.api_gateway import ApiGatewayResolver
app = ApiGatewayResolver(strip_prefixes=["/some-path"])
@app.get('/')
def my_func():
    return {'execution_status':'success'}
```

## Expected Behavior
Accessing `domain.com/some-path` would execute `my_func()`.

## Current Behavior
Accessing `domain.com/some-path` returns a 404.  Of note though, accessing `domain.com/some-path/` (note the trailing slash) *will* execute `my_func()`.

## Possible Solution
Add an conditional [here](https://github.com/awslabs/aws-lambda-powertools-python/blob/c8cf3ba2020e034e48543f2295a1aaf927d7177b/aws_lambda_powertools/event_handler/api_gateway.py#L549-L550) which will map an empty string to `"/"`.

## Steps to Reproduce (for bugs)
1. Set up the code example above
2. Set up ApiGateway to use a custom domain, and bind the Lambda to a custom path of `some-path`
3. Visit `https://CUSTOMDOMAIN/some-path` and `https://CUSTOMDOMAIN/some-path/` and note the different results

**Comment by michaelbrewer** [comment]
Nice one @BVMiko , i put in a review comment for adding a test case. 

Otherwise, i was thinking whether it should be `""` instead of `"/"`? Then this would work when the request is `/some-path`

```python
from aws_lambda_powertools.event_handler.api_gateway import ApiGatewayResolver
app = ApiGatewayResolver(strip_prefixes=["/some-path"])
@app.get("")
def my_func():
    return {'execution_status':'success'}
```

As the original request path of `/some-path` does different from `/some-path/`. Although `@app.get("")` does like strange.

**Comment by BVMiko** [comment]
I believe that the common standard among other routing frameworks is that all paths should have a leading slash, so `@api.get("/")` is usually the root path. Unfortunately the [Flask](https://flask.palletsprojects.com/en/2.0.x/api/#url-route-registrations) and [FastAPI](https://fastapi.tiangolo.com/tutorial/path-params/) docs don't explicitly state this, though all examples are written this way.  I did find [this Flask issue](https://github.com/pallets/flask/issues/2742) describing a related oversight of forgetting to enforce a leading slash though.

As for implementation, [Flask](https://github.com/pallets/flask/blob/afc13b9390ae2e40f4731e815b49edc9ef52ed4b/src/flask/blueprints.py#L392-L394) and [FastAPI](https://github.com/tiangolo/fastapi/blob/master/fastapi/routing.py#L519) both seem to do things differently by building and caching the full paths within each route. However [ExpressJS](https://github.com/expressjs/express/blob/master/lib/router/index.js#L300-L304), looks like it handles routing similar to the current ApiGatewayRouter though it does ensure a leading slash.

**Comment by michaelbrewer** [comment]
@BVMiko - agreed. But strip_prefix is supposed to remove the matching prefix from the request. 

And that this is on the context of "strip_prefixes" is "/custom-mapping", and the actual request is without a trailing "/" and would look like "/custom-mapping". So naturally I would except the resulting path to be "". 

Internally this was not happening due to the addition of a "/" for the starts with check. Because we wanted to allow for "/pay" and "/payment" strip prefixes to be set in any order. As "/pay" would be a partial match. It would remove before "/payment". 

So the ultimate solution could just be to reverse the order of the prefixes and remove the "/" we add in the starts with check.

**Comment by BVMiko** [comment]
@michaelbrewer - I updated the testcase and the docs to reflect the new explicit change; let me know if you'd like the wording modified.

**Comment by cakepietoast** [comment]
Fixed in [v1.20.2](https://github.com/awslabs/aws-lambda-powertools-python/releases/tag/v1.20.2).
